projects/components/src/datagrid/filters/datagrid-filter.ts

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { ClrDatagridFilterInterface } from '@clr/angular/data/datagrid/interfaces/filter.interface';
+import { ClrDatagridFilter } from '@clr/angular';
+import { Subject } from 'rxjs';
+import { ComponentRenderer } from '..';
+import { Input } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+
+/**
+ * Basic properties required by all the grid filters
+ */
+export interface FilterConfig<V> {
+    /**
+     * Used as a query field for server side filtering of the grid column
+     */
+    queryField?: string;
+
+    /**
+     * Value with which grid data can be filtered before initially being rendered
+     */
+    value?: V;
+}
+
+/**
+ * Extended by filter components used in {@link DatagridComponent}. Those components can only be used inside a
+ * clr-dg-filter component and are dynamically rendered by {@link ComponentRendererOutletDirective} using
+ * {@link GridColumn.filterRendererSpec}
+ * V is the type of filter input value that is passed into setValue method
+ * C extends FilterConfig<V> is configuration of a filter that contains queryField and a value of type V
+ */
+export abstract class DatagridFilter<V, C extends FilterConfig<V>>
+    implements ClrDatagridFilterInterface<unknown>, ComponentRenderer<C> {
+    formGroup: FormGroup;
+
+    protected constructor(filterContainer: ClrDatagridFilter) {
+        filterContainer.setFilter(this);
+    }
+
+    /**
+     * Contains configuration needed for a filter UI widget and also it's value.
+     * Assigned from {@link ComponentRendererOutletDirective#assignValue} after the filter component is created
+     */
+    private _config: C;
+    @Input() set config(val: C) {
+        this._config = val;
+        if (this._config.value) {
+            this.setValue(this._config.value);
+        }
+    }
+
+    /**
+     * Used by the getValue method in sub classes to format the FIQL string output
```

Comment:
Avoid comments that say `used by`. Does this need to be public? Having said that, a getter doesn't need additional comments if the setter already explains what it is.

Reply:
Yes, that method needs to be public. Added the explanation under the JS doc of setter and removed it from here.